test/e2e_env/multizone/producer/producer.go

```diff
@@ -0,0 +1,155 @@
+  ports:
+  - port: 80
+    name: main
+    targetPort: main
+    appProtocol: http
+`, k8sZoneNamespace, mesh, k8sZoneNamespace)
+
+		Expect(NewClusterSetup().
+			Install(NamespaceWithSidecarInjection(k8sZoneNamespace)).
+			Install(testserver.Install(
+				testserver.WithName("test-server"),
+				testserver.WithMesh(mesh),
+				testserver.WithNamespace(k8sZoneNamespace),
+				testserver.WithEchoArgs("echo", "--instance", "kube-test-server-2"),
+			)).
+			Install(YamlK8s(kubeServiceYAML)).
+			Setup(multizone.KubeZone2),
+		).To(Succeed())
+	})
+
+	AfterEachFailure(func() {
+		DebugUniversal(multizone.Global, mesh)
+		DebugKube(multizone.KubeZone1, mesh, k8sZoneNamespace)
+	})
+
+	E2EAfterEach(func() {
+		Expect(DeleteMeshResources(multizone.Global, mesh, meshtimeout_api.MeshTimeoutResourceTypeDescriptor)).To(Succeed())
+		Expect(DeleteMeshResources(multizone.Global, mesh, meshhttproute_api.MeshHTTPRouteResourceTypeDescriptor)).To(Succeed())
+	})
+
+	E2EAfterAll(func() {
+		Expect(multizone.KubeZone1.TriggerDeleteNamespace(k8sZoneNamespace)).To(Succeed())
+		Expect(multizone.KubeZone2.TriggerDeleteNamespace(k8sZoneNamespace)).To(Succeed())
+		Expect(multizone.Global.DeleteMesh(mesh)).To(Succeed())
+	})
+
+	It("should synced producer policy to other clusters", func() {
+		Expect(YamlK8s(fmt.Sprintf(`
+kind: MeshTimeout
+apiVersion: kuma.io/v1alpha1
+metadata:
+  name: to-test-server
+  namespace: %s
+  labels:
+    kuma.io/origin: zone
+    kuma.io/mesh: %s
+spec:
+  targetRef:
+    kind: Mesh
+  to:
+    - targetRef:
+        kind: MeshService
+        name: test-server
+      default:
+        http:
+          requestTimeout: 2s
+`, k8sZoneNamespace, mesh))(multizone.KubeZone2)).To(Succeed())
+
+		Eventually(func(g Gomega) {
+			out, err := k8s.RunKubectlAndGetOutputE(
+				multizone.KubeZone1.GetTesting(),
+				multizone.KubeZone1.GetKubectlOptions(Config.KumaNamespace),
+				"get", "meshtimeouts")
+			g.Expect(err).ToNot(HaveOccurred())
+			g.Expect(out).To(ContainSubstring(hash.HashedName(mesh, "to-test-server", Kuma2, k8sZoneNamespace)))
+		}).Should(Succeed())
+
+		Expect(YamlK8s(fmt.Sprintf(`
+kind: MeshTimeout
+apiVersion: kuma.io/v1alpha1
+metadata:
+  name: to-test-server
+  namespace: %s
+  labels:
+    kuma.io/origin: zone
+    kuma.io/mesh: %s
+spec:
+  targetRef:
+    kind: Mesh
+  to:
+    - targetRef:
+        kind: MeshService
+        name: test-server
+        namespace: random-ns-name
+      default:
+        http:
+          requestTimeout: 2s
+`, k8sZoneNamespace, mesh))(multizone.KubeZone2)).To(Succeed())
+
+		Eventually(func(g Gomega) {
+			out, err := k8s.RunKubectlAndGetOutputE(
+				multizone.KubeZone1.GetTesting(),
+				multizone.KubeZone1.GetKubectlOptions(Config.KumaNamespace),
+				"get", "meshtimeouts")
+			g.Expect(err).ToNot(HaveOccurred())
+			g.Expect(out).ToNot(ContainSubstring(hash.HashedName(mesh, "to-test-server", Kuma2, k8sZoneNamespace)))
+		}).Should(Succeed())
+	})
+}
```

Comment:
Can we check if it's actually applied on the client? instead of just synced? That's more E2E test